src/cursor/cur_version.c

```diff
@@ -0,0 +1,362 @@
+/*
+ * __curversion_next --
+ *     WT_CURSOR->next method for version cursors. The next function will position the cursor on the
+ *     next update of the key it is positioned at. We traverse through updates on the update chain,
+ *     then the ondisk value, and finally from the history store.
+ */
+static int
+__curversion_next(WT_CURSOR *cursor)
+{
+    WT_CURSOR *hs_cursor, *table_cursor;
+    WT_CURSOR_BTREE *cbt;
+    WT_CURSOR_VERSION *version_cursor;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    WT_TIME_WINDOW tw;
+    WT_UPDATE *upd;
+
+    version_cursor = (WT_CURSOR_VERSION *)cursor;
+    hs_cursor = version_cursor->hs_cursor;
+    table_cursor = version_cursor->table_cursor;
+    cbt = (WT_CURSOR_BTREE *)table_cursor;
+    CURSOR_API_CALL(cursor, session, next, NULL);
+
+    ret = WT_NOTFOUND;
+
+    /* The cursor should be positioned. */
+    if (!F_ISSET(cursor, WT_CURSTD_KEY_INT))
+        goto err;
+
+    upd = version_cursor->next_upd;
+
+update_chain:
+    if (upd != NULL && !F_ISSET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED)) {
+        /*
+         * If the update is an aborted update, we want to skip to the next update immediately or get
+         * the ondisk value if the update is the last one in the update chain.
+         */
+        if (upd->txnid == WT_TXN_ABORTED) {
+            upd = upd->next;
+            if (upd != NULL)
+                goto update_chain;
+            else {
+                F_SET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED);
+                goto disk_image;
+            }
+        }
+        /*
+         * If the update is a tombstone, we still want to record the stop information but we also
+         * need traverse to the next update to get the full value. If the tombstone was the last
+         * update in the update list, retrieve the ondisk value.
+         */
+        version_cursor->upd_txnid = upd->txnid;
+        version_cursor->upd_durable_stop_ts = upd->durable_ts;
+        version_cursor->upd_stop_ts = upd->start_ts;
+        if (upd->type == WT_UPDATE_TOMBSTONE) {
+            upd = upd->next;
+            if (upd == NULL) {
+                F_SET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED);
+                goto disk_image;
+            }
+        }
+
+        /*
+         * Set the version cursor's key, which contains all the record metadata for that particular
+         * version of the update.
+         */
+        __wt_cursor_set_key(cursor, upd->txnid, upd->durable_ts, upd->start_ts,
+          version_cursor->upd_txnid, version_cursor->upd_durable_stop_ts,
+          version_cursor->upd_stop_ts, upd->type, upd->prepare_state, upd->flags,
+          WT_VERSION_UPDATE_CHAIN);
+
+        /*
+         * Set the version cursor's value to be the update value. If the update is a modify,
+         * reconstruct the value.
+         */
+        if (upd->type != WT_UPDATE_MODIFY) {
+            __wt_upd_value_assign(cbt->upd_value, upd);
+            ret = 0;
```

Comment:
Why you need to set ret = 0?

Reply:
At the start of the function, I initialised ret = WT_NOTFOUND, so here I am thinking that we found a value.

Reply:
This is against the convention. We should do it in the opposite way.

Reply:
Sorry, I will change it back.